repository/src/test/java/org/pentaho/platform/repository2/unified/jcr/DefaultDeleteHelperTest.java

```diff
@@ -385,6 +388,17 @@ protected List<String> getUserList() {
     final List<RepositoryFile> deletedFilesAdmin = defaultDeleteHelper.getAllDeletedFiles( session, pentahoJcrConstants );
     assertNotNull( deletedFilesAdmin );
     assertEquals( 3, deletedFilesAdmin.size() );
+
+    // as admin all user trash access disabled
+    try ( MockedStatic<PentahoSystem> mockedPentahoSystem = mockStatic( PentahoSystem.class ) ) {
+      mockedPentahoSystem.when(
+        () -> PentahoSystem.getSystemSetting( "adminAccessAllUsersTrash", "false" ) )
+        .thenReturn( "false" );
+      admin[0] = true;
+      final List<RepositoryFile> deletedFilesAdminNoAccess = defaultDeleteHelper.getAllDeletedFiles( session, pentahoJcrConstants );
+      assertNotNull( deletedFilesAdminNoAccess );
+      assertEquals( 1, deletedFilesAdminNoAccess.size() );
+    }
```

Comment:
Please check if it is possible to add a test for the scenario  would be great if there could be a test for when adminAccessAllUsersTrash is set to true;

Reply:
Added, also refactored the unit tests to have one test method per test case, while reusing the setup instructions code